test/Microsoft.SqlTools.ServiceLayer.UnitTests/QueryExecution/Execution/ServiceIntegrationTests.cs

```diff
@@ -37,6 +37,24 @@ public void ExecuteDocumentStatementTest()
         }
 
         [Fact]
+        public void ExecuteDocumentStatementSameLine()
+        {
+            var statement1 = Constants.StandardQuery;
+            var statement2 = "SELECT * FROM sys.databases";
+            string query = string.Format("{0};{1}", statement1, statement2);
+            var workspaceService = GetDefaultWorkspaceService(query);
+            var queryService = new QueryExecutionService(null, workspaceService);
+
+            // If a line has multiple statements and the cursor is somewhere in the second one
+            var cursorColumn = statement1.Length + 1;
+            var queryParams = new ExecuteDocumentStatementParams { OwnerUri = Constants.OwnerUri, Line = 0, Column = cursorColumn };
+            var queryText = queryService.GetSqlText(queryParams);
+
+            // The text should match the second statement
+            Assert.Equal(queryText, statement2);
```

Comment:
A couple additional checks that should be quick to add:
- Verify index 0 returns statement 1
- Verify index statement1.Length (which is just to the left of the `;`) returns statement 1
- Verify the end of the text gives statement 2

I think this should be quick and cover edge cases that might come up if someone chooses to change it and accidentally add a regression in the future.